sklearn/calibration.py

```diff
@@ -885,29 +886,34 @@ def _sigmoid_calibration(
     T = np.zeros_like(y, dtype=np.float64)
     T[y > 0] = (prior1 + 1.0) / (prior1 + 2.0)
     T[y <= 0] = 1.0 / (prior0 + 2.0)
-    T1 = 1.0 - T
 
-    def objective(AB):
-        # From Platt (beginning of Section 2.2)
-        P = expit(-(AB[0] * F + AB[1]))
-        loss = -(xlogy(T, P) + xlogy(T1, 1.0 - P))
-        if sample_weight is not None:
-            return (sample_weight * loss).sum()
-        else:
-            return loss.sum()
+    bin_loss = HalfBinomialLoss()
 
-    def grad(AB):
-        # gradient of the objective function
-        P = expit(-(AB[0] * F + AB[1]))
-        TEP_minus_T1P = T - P
-        if sample_weight is not None:
-            TEP_minus_T1P *= sample_weight
-        dA = np.dot(TEP_minus_T1P, F)
-        dB = np.sum(TEP_minus_T1P)
-        return np.array([dA, dB])
+    def loss_grad(AB):
+        l, g = bin_loss.loss_gradient(
+            y_true=T,
+            raw_prediction=-(AB[0] * F + AB[1]),
```

Comment:
The losses need contiguous arrays. Are we sure the one passed to raw_prediction is?

Reply:
If F is non-contiguous, then I don’t know what `3 * F + 1` is contiguous or not.

Reply:
```python
predictions = column_or_1d(predictions)

F = predictions
```

According to the definition in the function `column_or_1d`, it seems that this is set to a contiguous array.

https://github.com/scikit-learn/scikit-learn/blob/989d29946a197eedde3c30f71ea4498e6642ec05/sklearn/utils/validation.py#L1219-L1243